provider/block_devices_plug.py

```diff
@@ -261,15 +262,23 @@ def _create_devices(self, images, pci_bus={"aobject": "pci.0"}):
                     self._hotplugged_devs[img].insert(-1, dev)
                     HOTPLUGGED_HBAS[img] = dev
 
-    def _plug(self, plug_func, monitor):
+    def _plug(self, plug_func, monitor, action):
         end = time.time() + self.ACQUIRE_LOCK_TIMEOUT
         while time.time() < end:
             try:
-                return plug_func(monitor)
+                return plug_func(
+                    monitor
+                ) if action == UNPLUG else plug_func(
+                    monitor, self._qemu_version
+                )
```

Comment:
```suggestion
                return plug_func(
                    monitor
                ) if plug_func.__name__ ==  'unplug' else plug_func(
                    monitor, self._qemu_version
                )
```
I think we could not add external argument for it by using `__name__`
BTW， your solution is look good to me since this is prviate method.